modules/cart/src/CartManager.php

```diff
@@ -123,6 +124,7 @@ public function addOrderItem(OrderInterface $cart, OrderItemInterface $order_ite
     }
 
     if ($save_cart) {
+      $this->resetCheckoutFlow($cart);
```

Comment:
Let's perform the reset before the if(), that way the step is reset even if saving the cart is skipped.